src/with-cache.ts

```diff
@@ -0,0 +1,75 @@
+import * as core from "@actions/core";
+import * as cache from "@actions/cache";
+import type { CacheKeys } from "./get-cache-keys";
+
+export interface CoreDelegate {
```

Comment:
Use mock modules for `core` and `cache` to remove the notion of delegation here.